aws/table_aws_resource_explorer_search.go

```diff
@@ -0,0 +1,147 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/resourceexplorer2"
+	"github.com/aws/aws-sdk-go-v2/service/resourceexplorer2/types"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAWSResourceExplorerSearch(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_resource_explorer_search",
+		Description: "AWS Resource Explorer Search",
+		List: &plugin.ListConfig{
+			Hydrate: awsResourceExplorerSearch,
+			KeyColumns: plugin.KeyColumnSlice{
+				{Name: "query", Require: plugin.Optional, CacheMatch: "exact"},
+				{Name: "region", Require: plugin.Required},
```

Comment:
In an effort to make this table a bit easier, I think we should update the search table to work as follows:
- The table should have no required key columns, and we remove `region` as a key column entirely (but keep `query` and `view_arn`
- Queries for this table that do not specify any key columns, e.g., `select * from aws_resource_explorer_search`, should still work and follow this sequence
  - List all indexes and look for an aggregator index
    - If there's no aggregator index in the account, return an error letting the user know there's no aggregator
  - Connect to the region where the aggregator index exists
  - Look for a default view in the aggregator region
    - If there's no default view, return an error letting the user know there's no default view
    - Note: If this isn't possible due to https://github.com/aws/aws-sdk-go-v2/issues/1916, we should skip this step for now and just make the search call
  - Make the search API call
  
 This design has been largely modeled off of how AWS handles activating [unified search](https://docs.amazonaws.cn/en_us/resource-explorer/latest/userguide/using-unified-search.html):
> To enable including your account's resources in the search results for unified search from any Amazon console, you must complete the following steps:
> 1. [Activate Amazon Resource Explorer in one or more Amazon Web Services Regions in your account.](https://docs.amazonaws.cn/resource-explorer/latest/userguide/manage-service-register.html)
> 2. [Register one Region to contain the aggregator index.](https://docs.amazonaws.cn/resource-explorer/latest/userguide/manage-aggregator-region.html)
> 3. [Create a default view in the Region with the aggregator index.](https://docs.amazonaws.cn/resource-explorer/latest/userguide/manage-views.html)

If a user does pass a `view_arn`, then we should extract the region out of the view ARN, connect to that region, and pass that `view_arn` in the input params.

Other notes:
- AWS has a 500 requests/month limit on single region indexes and 10,000 requests/month limit on aggregator region indexes, so we strongly prefer searching on the aggregator indexes.
- AWS in general returns 401 UnauthorizedException error codes for any error, e.g., Resource Explorer disabled, no index, no default view, insufficient IAM permissions, so we should do our best in the code to wrap these errors and provide better error messages when possible.